File: src/features/auth/dto/register.dto.ts
import { IsBoolean, IsEmail, IsEnum, IsOptional, IsString, Matches } from 'class-validator';
import { STRONG_PASSWORD_MESSAGE, STRONG_PASSWORD_REGEX } from './password-rules';
import { Role } from '../../../common/roles';

export class RegisterDto {
  @IsOptional()
  @IsString()
  firstName?: string;

  @IsOptional()
  @IsString()
  lastName?: string;

  @IsString()
  username: string;

  @IsEmail()
  email: string;

  @IsString()
  @Matches(STRONG_PASSWORD_REGEX, { message: STRONG_PASSWORD_MESSAGE })
  password: string;

  @IsOptional()
  @IsString()
  organizationId?: string;

  @IsOptional()
  @IsString()
  organizationName?: string;

  @IsOptional()
  @IsEnum(Role)
  role?: Role;

  @IsOptional()
  @IsString()
  inviteToken?: string;

  @IsBoolean()
  legalAccepted: boolean;

  @IsOptional()
  @IsBoolean()
  orgLegalReconfirm?: boolean;
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/auth/RegisterDto.java
package com.mytegroup.api.dto.auth;

import com.mytegroup.api.common.enums.Role;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RegisterDto {
    
    private String firstName;
    
    private String lastName;
    
    @NotBlank(message = "Username is required")
    private String username;
    
    @NotBlank(message = "Email is required")
    @Email(message = "Email must be valid")
    private String email;
    
    @NotBlank(message = "Password is required")
    @Pattern(regexp = PasswordRules.STRONG_PASSWORD_REGEX, message = PasswordRules.STRONG_PASSWORD_MESSAGE)
    private String password;
    
    private String orgId;
    
    private String organizationName;
    
    private Role role;
    
    private String inviteToken;
    
    @NotNull(message = "Legal acceptance is required")
    private Boolean legalAccepted;
    
    private Boolean orgLegalReconfirm;
    
    public void setEmail(String email) {
        this.email = email != null ? email.toLowerCase().trim() : null;
    }
    
    public void setUsername(String username) {
        this.username = username != null ? username.trim() : null;
    }
    
    public void setFirstName(String firstName) {
        this.firstName = firstName != null ? firstName.trim() : null;
    }
    
    public void setLastName(String lastName) {
        this.lastName = lastName != null ? lastName.trim() : null;
    }
    
    public void setOrganizationName(String organizationName) {
        this.organizationName = organizationName != null ? organizationName.trim() : null;
    }
}
